### PR TITLE
[EC-451] Org Admin Refresh Permissions Refactor

### PR DESCRIPTION
**Pull request by shane-melton**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

#3252 refactored the organization routing permissions which were also modified as part of the org admin refresh. This PR properly brings those changes into `feature/org-admin-refresh` branch in order to account for the new tabs.

## Code changes

- **apps/web/src/app/organizations/billing/organization-billing-routing.module.ts:** Use the new `canAccessBillingTab` callback and remove redundant guards for child routes
- **apps/web/src/app/organizations/layouts/organization-layout.component.ts:** Use the new `canAccess<Tab>` callbacks
- **apps/web/src/app/organizations/navigation-permissions.ts:** Add new `canAccess<Tab>` callbacks for the new tabs added as part of the org admin refresh. And remove unused callbacks.
- **apps/web/src/app/organizations/reporting/organization-reporting-routing.module.ts:** Use the new `canAccessReportingTab` callback and remove redundant guards for child routes

## Before you submit

```
- [X] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/web/src/app/organizations/navigation-permissions.ts [hunk 1]
```diff
@@ -25,5 +37,12 @@ export function canAccessManageTab(org: Organization): boolean {
...
 
 export function canAccessOrgAdmin(org: Organization): boolean {
-  return canAccessToolsTab(org) || canAccessSettingsTab(org) || canAccessManageTab(org);
+  return (
+    canAccessMembersTab(org) ||
+    canAccessGroupsTab(org) ||
+    canAccessReportingTab(org) ||
+    canAccessBillingTab(org) ||
+    canAccessSettingsTab(org) ||
+    canAccessManageTab(org)
```
Comment (eliykat) [inline apps/web/src/app/organizations/navigation-permissions.ts:46]: It looks like the Manage tab has been removed as part of the refresh, if so then you can delete this function. (There is a reference to it in the `bit-web` `organization-routing.module.ts`, but that seems safe to remove given that all child routes are guarded and that'll be changed as part of this refresh anyway.)

File: apps/web/src/app/organizations/navigation-permissions.ts [hunk 2]
```diff
@@ -1,7 +1,19 @@
 import { Organization } from "@bitwarden/common/models/domain/organization";
 
-export function canAccessToolsTab(org: Organization): boolean {
-  return org.canAccessImportExport || org.canAccessReports;
+export function canAccessMembersTab(org: Organization): boolean {
```
Comment (eliykat) [inline apps/web/src/app/organizations/navigation-permissions.ts:3]: You need to update the route to use this new function (in the OSS `organization-routing.module.ts`)

File: apps/web/src/app/organizations/navigation-permissions.ts [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+  return org.canManageUsers || org.canManageUsersPassword;
+}
+
+export function canAccessGroupsTab(org: Organization): boolean {
```
Comment (eliykat) [inline apps/web/src/app/organizations/navigation-permissions.ts:7]: You need to update the route to use this new function (in the OSS `organization-routing.module.ts`)

**Review by eliykat — CHANGES_REQUESTED** [review]
Thanks for bringing this across to the feature branch! A few comments below.

File: apps/web/src/app/organizations/navigation-permissions.ts (hunk 1 above)
Comment (shane-melton) [reply apps/web/src/app/organizations/navigation-permissions.ts:46]: Yep it should be removed, but I had left it due to the left over reference in `bit-web`. But you make a good point that it's already guarded and will be changed as part of #3251 anyways.

File: apps/web/src/app/organizations/navigation-permissions.ts (hunk 2 above)
Comment (shane-melton) [reply apps/web/src/app/organizations/navigation-permissions.ts:3]: Yep, good catch!

**Review by eliykat — APPROVED** [review]

**Pull request merged**